spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/CompletionConfiguration.java

```diff
@@ -46,81 +49,89 @@
 	private ApplicationConfigurationMetadataResolver metadataResolver;
 
 	@Bean
-	public StreamCompletionProvider streamCompletionProvider() {
+	@ConditionalOnMissingBean
+	public StreamDefinitionService streamDefinitionService() {
+		return new DefaultStreamDefinitionService();
```

Comment:
Move to main auto-config

Reply:
Moving this into main auto-config involves some other refactoring since the completion configuration is for both stream/task.